includes/Modules/AdSense.php

```diff
@@ -551,17 +495,12 @@ function( Google_Service_AdSense_Alert $alert ) {
 					return new WP_Error( 'missing_required_param', sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'clientID' ), array( 'status' => 400 ) );
 				}
 				return function() use ( $data ) {
-					$option                  = $this->get_settings()->get();
-					$option['setupComplete'] = true;
-					$option['clientID']      = $data['clientID'];
-					$option['useSnippet']    = isset( $option['useSnippet'] ) ? true : $data['useSnippet'];
-
-					// Set useSnippet explicitly using $data param, otherwise default to true if not set in option.
-					if ( isset( $data['useSnippet'] ) ) {
-						$option['useSnippet'] = $data['useSnippet'];
-					}
+					$option = array_merge(
+						$data->data,
+						array( 'setupComplete' => true )
```

Comment:
We should maybe keep this as it is now, i.e. only look at the specific "white-listed" fields, not merge any given data.

Reply:
Yeah, that's probably a good idea. What do you think about making the second parameter of `merge` a whitelist of keys?

Reply:
I think we should keep the method simple as it is. Whether it should restrict further than what the option generally allows should remain responsibility of the caller.